setup.py

```diff
@@ -111,7 +111,6 @@
           'cffi==1.11.2',
           'awscli>=1.11.91',
           'boto3',
-          'moto==1.3.4',
```

Comment:
moto should stay here (and moto should be moved into `ESSENTIAL TESTING-RELATED PACKAGES`). These are the requirements that get bundled in `pip install allennlp`, and `allennlp test-install` will fail out of the box if it doesn't install moto (since it's imported in the test https://github.com/joelgrus/allennlp/blob/fc83493e4c46e9bfc8e4f7bd5a5347f01673b78d/allennlp/tests/common/file_utils_test.py#L10)

Reply:
Gotcha. So wouldn't that be true of `pytest`, `flaky`, and `responses` as well, since they are all imported by modules in `tests/`?

Reply:
In which case, everything under `tests_require` would already be under `install_requires`, right?

Reply:
yup, `pytest`, `flaky`, and `responses` are already in `install_requires`.

> In which case, everything under tests_require would already be under install_requires, right?

Excepting jupyter (some tests run notebooks, but we skip those by default in `test-install`), yes. So maybe we should move jupyter to "ESSENTIAL TESTING-RELATED PACKAGES" as well.

The key is that for jupyter, we want to make clear that it's needed to run the tests (for now, I think there are efforts to rework tutorials into vanilla md), but also not install it by default since it's massive.

Reply:
Ah, got it! Thanks for the clarification. I'll need to make a couple of adjustments